presto-main/src/main/java/com/facebook/presto/operator/scalar/MergingDigest.java

```diff
@@ -767,155 +660,136 @@ public double quantile(double q) {
     }
 
     @Override
-    public int centroidCount() {
+    public int centroidCount()
+    {
         mergeNewValues();
-        return lastUsedCell;
+        return activeCentroids;
     }
 
     @Override
-    public Collection<Centroid> centroids() {
-        // we don't actually keep centroid structures around so we have to fake it
+    public Collection<Centroid> centroids()
+    {
         compress();
-        return new AbstractCollection<Centroid>() {
+        return new AbstractCollection<Centroid>()
+        {
             @Override
-            public Iterator<Centroid> iterator() {
-                return new Iterator<Centroid>() {
-                    int i = 0;
+            public Iterator<Centroid> iterator()
+            {
+                return new Iterator<Centroid>()
+                {
+                    int i;
 
                     @Override
-                    public boolean hasNext() {
-                        return i < lastUsedCell;
+                    public boolean hasNext()
+                    {
+                        return i < activeCentroids;
                     }
 
                     @Override
-                    public Centroid next() {
+                    public Centroid next()
+                    {
                         Centroid rc = new Centroid(mean[i], (int) weight[i], data != null ? data.get(i) : null);
                         i++;
                         return rc;
                     }
 
                     @Override
-                    public void remove() {
+                    public void remove()
+                    {
                         throw new UnsupportedOperationException("Default operation");
                     }
                 };
             }
 
             @Override
-            public int size() {
-                return lastUsedCell;
+            public int size()
+            {
+                return activeCentroids;
             }
         };
     }
 
     @Override
-    public double compression() {
+    public double compression()
+    {
         return publicCompression;
     }
 
     @Override
-    public int byteSize() {
+    public int byteSize()
+    {
         compress();
-        // format code, compression(float), buffer-size(int), temp-size(int), #centroids-1(int),
+        // format code, type code, min (double), max (double), compression(double), #centroids(int),
```

Comment:
I don't feel that the new comment is clearer than existing one. Why do we need to change this? And why it seems that the types changed to something bigger while the size actually is reduced by 2?

Reply:
This comment was originally wrong. It was a comment on the "small" serialization version, which we have discarded. I tried making the comment a bit clearer:
```
        // format (byte), type (byte), min (double), max (double), compression(double), total weight(double) #centroids(int),
        // then weight[] (double array) and getMean[] (double array)
```
I also documented the serialization format that we'll be using [here](https://bit.ly/2MZhTks).